README_sourcetree.md

```diff
@@ -0,0 +1,81 @@
+## Source tree layout
```

Comment:
I would suggest moving this file to `docs/sourcetree.md` and reference it in the `docs/README.md` TOC